enhancements/cgu-precaching-additional-images.md

```diff
@@ -0,0 +1,605 @@
+The ConfigMap will be deprecated starting with this release.
+
+An example of the proposed pre-caching configuration CR is shown below.
+
+```yaml
+apiVersion: ran.openshift.io/v1alpha1
+kind: PreCachingConfig
+metadata:
+  name: foobar
+spec:
+  overrides:
+    platformImage: quay.io/openshift-release-dev/ocp-release@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    operatorsIndexes:
+      - registry.example.com:5000/custom-redhat-operators:1.0.0
+    operatorsPackagesAndChannels:
+      - local-storage-operator: stable
+      - ptp-operator: stable
+      - sriov-network-operator: stable
+  spaceRequired: 30Gi
+  excludePrecachePatterns:
+    - aws
+    - vsphere
+  additionalImages:
+    - quay.io/foobar/application1@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    - quay.io/foobar/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+    - quay.io/foobar/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+```
+
+It is important to note that all the fields in the CR are optional. Also worthy to note that the
+`platformImage`, `operatorsIndexes` and `operatorsPackagesAndChannels` fields have been collectively
+grouped under the `overrides` object. This is done to indicate to the user that these values need not
+be specified as they are automatically derived from the policies pertaining to the managed clusters.
+Furthermore, it also provides the context of the user overriding the default TALM functionality. Should
+there be new fields in the future which are automatically computed by TALM, these can be added here as
+well. The CR also contains 2 new fields, namely `spaceRequired` and `additionalImages`.
+
+The `additionalImages` field allows the user to specify the list of additional images they desire to be
+pre-cached. This field specifically addresses the topic of the EP. The `spaceRequired` field allows the
+user to specify the minimum required disk space (specified in Gigabytes) on the cluster. TALM defines a
+default value for OpenShift related images if the parameter is unspecified. Prior to pulling the images,
+the pre-caching job performs a check to verify whether there is enough disk space to avoid triggering
+kubelet to perform garbage collection (GC) and thus mitigate the benefit of the pre-caching job.
+The `spaceRequired` value should take into account the disk space required by both OpenShift related images
+as well as the additional user images.
+
+TALM reports the pre-caching status in the CGU as shown below. Although the `platformImage`, `operatorsIndexes`
+and `operatorsPackagesAndChannels` fields are nested under the `PreCachingConfig.spec.overrides` object, it
+is important to note that they will still be shown as a flat json object under `status.precaching` in the CGU CR.
+
+```json
+{
+...
+  "status": {
+    "conditions": [...],
+    "precaching": {
+        "clusters": ["sno1", "sno2"],
```

Comment:
this has been removed.

Reply:
Thanks for bringing that to my attention. I have removed this line.